homework-submission/semih-week2/form.js

```diff
@@ -0,0 +1,90 @@
+
+class TodoForm extends React.Component {
+    constructor(){
+        super();
+        this.state = {
+            todos : [
+                {
+                    "id" : 1,
+                    "description" : "Get out of bed",
+                    "deadline" : "2017-09-11",
+                    "done" : true
+                },
+                {
+                    "id" : 2,
+                    "description" : "Brush teeth",
+                    "deadline" : "2017-09-10",
+                    "done" : false
+                },
+                {
+                    "id" : 3,
+                    "description" : "Eat breakfast",
+                    "deadline" : "2017-09-09",
+                    "done" : false
+                },
+                {
+                    "description" : "add some damn css :)",
+                    "done" : false
+                },
+            ]
+            
+        };
+    }
+
+    createToDo = () => {
+        const todos = this.state.todos;
+        todos.push({"description":"seriously, add some css! :)"});
+        this.setState({todos:todos});    
+    }
+    removeToDo = () => {
+        const todos = this.state.todos;
+        todos.pop();
+        this.setState({todos:todos});    
+    }  
+    submitFormHandler = event => {
+        event.preventDefault();
+        const newItem = this.refs.todo.value
+        const todos = this.state.todos;
+        todos.push({"description":newItem,"done":false});
+        this.setState({todos:todos});   
+    }
+
+    markAsDone = () => {
+        event.preventDefault();
+        const markasdoneIndex = this.refs.markasdone.value-1
+        const todos = this.state.todos;
+        todos[markasdoneIndex].done = true;
+        this.setState({todos:todos});
+    }
+        
+    render(){
+        const todos = this.state.todos;
+        return (
```

Comment:
Perhaps some splitting the content will make this `jsx` content easier to be read.